Classes/Task/IndexQueueWorkerTaskAdditionalFieldProvider.php

```diff
@@ -132,17 +133,18 @@ public function saveAdditionalFields(
         array $submittedData,
         AbstractTask $task
     ) {
-        $this->isTaskInstanceofIndexQueueWorkerTask($task);
-
-        $task->setSite(GeneralUtility::makeInstance(Site::class, $submittedData['site']));
-        $task->setDocumentsToIndexLimit($submittedData['documentsToIndexLimit']);
-        $task->setForcedWebRoot($submittedData['forcedWebRoot']);
+        if ($this->isTaskInstanceofIndexQueueWorkerTask($task)) {
```

Comment:
I would propose to keep the usages of isTaskInstanceofIndexQueueWorkerTask as it was because otherwise we just introduce another level maybe it is an option to rename isTaskInstanceofIndexQueueWorkerTask in checkIsTaskInstanceofIndexQueueWorkerTask ?